awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,218 @@
+                          'num_hosts': smart_inv.hosts.count(),
+                          'num_sources': smart_inv.inventory_sources.count()
+                          }
+    return counts
+
+
+@register('projects_by_scm_type')
+def projects_by_scm_type(since):
+    counts = dict(
+        (t[0] or 'manual', 0)
+        for t in models.Project.SCM_TYPE_CHOICES
+    )
+    for result in models.Project.objects.values('scm_type').annotate(
+        count=Count('scm_type')
+    ).order_by('scm_type'):
+        counts[result['scm_type'] or 'manual'] = result['count']
+    return counts
+
+@register('instance_info')   # maybe integrate this with job_instance_counts?
+def instance_info(since):
+    info = {}
+    instances = models.Instance.objects.values_list('hostname').annotate().values('uuid', 'version', 'capacity', 'cpu', 'memory', 'managed_by_policy', 'hostname')
+    for instance in instances:
+        info = {'uuid': instance['uuid'],
+                'version': instance['version'],
+                'capacity': instance['capacity'],
+                'cpu': instance['cpu'],
+                'memory': instance['memory'],
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):         #TODO: all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+    job_statuses = models.UnifiedJob.objects.values_list(
+        'execution_node', 'status').annotate(job_status=Count('status'))
+    for job in job_statuses:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+# See if we can group by status and type inside the node grouping.  ^^
+
+    # 'jobs_running': models.UnifiedJob.objects.filter(execution_node=instance.hostname, status__in=('running', 'waiting',)).count(), # jobs in running & waiting state
+    # 'jobs_total': models.UnifiedJob.objects.filter(execution_node=instance.hostname).count(),
+    # ^^ try to get jobs_running and jobs_total per instance.  and uuid if easily possible  
+
+    return counts
+
+
+# Copies Job Events from db to a .csv to be shipped
+def copy_events_table(path='/tmp/'):
+    events_file = os.path.join(path, 'events_table.csv')
+    try:
+        write_data = open(events_file2, 'w')
```

Comment:
`NameError`

Reply:
that was a troubleshooting artifact.  I am abstracting a `copy_table()` method out and using it for an event and job table drops.  and need to add since, rather than hard coding 24 hours.